### PR TITLE
Fix false positives for interpolation in keyframes-name-pattern

### LINKED ISSUE #5997: Fix false positives for interpolation in keyframes-name-pattern
### What steps are needed to reproduce the bug?

test.scss

```scss
$frames: ("one", "two", "three");

@each $name in $frames {
  @keyframes frame-#{$name} {
    /* CSS ... */
  }
}
```

```
$ npx stylelint test.scss

test.scss
 4:14  ✖  Expected keyframe name to be kebab-case  keyframes-name-pattern
```

The name is kebab case, but due to the string interpolation, the line is flagged as an error.

This was originally file at https://github.com/stylelint-scss/stylelint-config-standard-scss/issues/16 but I was instructed to move it here.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard-scss"
}
```

### How did you run Stylelint?

npx stylelint test.scss

### Which version of Stylelint are you using?

`14.6.1`

### What did you expect to happen?

No problems to be reported.

### What actually happened?

```
test.scss
 4:14  ✖  Expected keyframe name to be kebab-case  keyframes-name-pattern
```

### Does the bug relate to non-standard syntax?

Yes it uses SCSS string interpolation.

### Proposal to fix the bug

_No response_

### PR DESCRIPTION
**Pull request by a-falkonia**
> Which issue, if any, is this issue related to?

Closes #5997 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/keyframes-name-pattern/index.js [hunk 1]
```diff
@@ -38,6 +39,10 @@ const rule = (primary) => {
 				return;
 			}
 
+			if (!isStandardSyntaxKeyframesName(value)) {
```
Comment (jeddy3) [inline lib/rules/keyframes-name-pattern/index.js:42]: Let's move this to before the first conditionally as we normally check that things are standard before any other operation.

File: lib/utils/__tests__/isStandardSyntaxKeyframesName.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const isStandardSyntaxKeyframesName = require('../isStandardSyntaxKeyframesName');
+
+describe('isStandardSyntaxKeyframesName', () => {
+	it('standard name', () => {
+		expect(isStandardSyntaxKeyframesName('@keyframes slidein {}')).toBeTruthy();
```
Comment (jeddy3) [inline lib/utils/__tests__/isStandardSyntaxKeyframesName.test.js:7]:
```suggestion
		expect(isStandardSyntaxKeyframesName('slidein')).toBeTruthy();
```

As the util is checking the [keyframes-name](https://drafts.csswg.org/css-animations/#typedef-keyframes-name), and not the whole `@keyframe`.

We'll need to update the other tests too.

File: lib/utils/isStandardSyntaxKeyframesName.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Check whether a keyframes name is standard
+ *
+ * @param {string} keyframesName
+ * @returns {boolean}
+ */
+
+module.exports = function (keyframesName) {
+	// SCSS or Less interpolation
```
Comment (jeddy3) [inline lib/utils/isStandardSyntaxKeyframesName.js:13]:
```suggestion
```

We can loose the comment. (The `hasInterpolation` checks for more than just Less or Sass interpolation).

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@a-falkonia Thanks for opening the pull request. It's looking good.

I've requested some minor changes.

**Comment by a-falkonia** [comment]
@jeddy3 Thank you for your review!
I've made the required changes, hope everything looks good now.

File: lib/utils/__tests__/isStandardSyntaxKeyframesName.test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const isStandardSyntaxKeyframesName = require('../isStandardSyntaxKeyframesName');
+
+describe('isStandardSyntaxKeyframesName', () => {
+	it('standard name', () => {
+		expect(isStandardSyntaxKeyframesName('slidein {}')).toBeTruthy();
```
Comment (ybiquitous) [inline lib/utils/__tests__/isStandardSyntaxKeyframesName.test.js:7]: [suggest] `toBe(true)` / `toBe(false)` are more accurate than `toBeTruthy()` / `toBeFalsy()`. E.g.

```suggestion
		expect(isStandardSyntaxKeyframesName('slidein {}')).toBe(true);
```

File: lib/utils/isStandardSyntaxKeyframesName.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+const hasInterpolation = require('./hasInterpolation');
+
+/**
+ * Check whether a keyframes name is standard
+ *
+ * @param {string} keyframesName
+ * @returns {boolean}
+ */
+
```
Comment (ybiquitous) [inline lib/utils/isStandardSyntaxKeyframesName.js:11]: [trivial] Can you please remove the empty line for consistency?

```suggestion
```

File: lib/utils/isStandardSyntaxKeyframesName.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+module.exports = function (keyframesName) {
```
Comment (ybiquitous) [inline lib/utils/isStandardSyntaxKeyframesName.js:12]: [suggest] Can you please give a function name for debuggability (avoiding anonymous functions)?

```suggestion
module.exports = function isStandardSyntaxKeyframesName(keyframesName) {
```

**Review by ybiquitous — COMMENTED** [review]
@a-falkonia Thanks for creating the PR!

I've left some refactoring suggestions, but the change looks almost good. 👍🏼

**Comment by a-falkonia** [comment]
@ybiquitous Thank you! I've made the updates.

**Review by ybiquitous — APPROVED** [review]
Thank you! LGTM 👍🏼

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@a-falkonia Thanks for making the changes. All most there.

I've requested one set of minor changes to the tests.

**Comment by a-falkonia** [comment]
@jeddy3 So so sorry for missing this. All the blocks should be removed now.

**Review by jeddy3 — APPROVED** [review]
Thank you!

**Comment by jeddy3** [comment]
Changelog entry added:

> - Fixed: `keyframes-name-pattern` false positives for interpolation ([#6265](https://github.com/stylelint/stylelint/pull/6265)).

**Pull request merged**